### ISSUE TITLE
Single argument separated by '&' has an ugly line break for typescript parser

### ISSUE THREAD

**Issue by faheem-akhtar** [typescript-eslint/typescript-eslint#1219]
**Source code**

```ts
// functional component with ugly line break
export const MyLongNamedReactFunctionalComponent1: FunctionComponent<ALongNamedInterface1 & ALongNamedInterface2> = (props) => {}

// functional component with valid line break
export const MyLongNamedReactFunctionalComponent2: FunctionComponent<ALongNamedInterface1 | ALongNamedInterface2> = (props) => {}

// functional component with valid line break
export const MyLongNamedReactFunctionalComponent3: FunctionComponent<ALongNamedInterface1, ALongNamedInterface2> = (props) => {}
```

**Output**
```ts
// functional component with ugly line break
export const MyLongNamedReactFunctionalComponent1: FunctionComponent<ALongNamedInterface1 &
  ALongNamedInterface2> = props => {};

// functional component with valid line break
export const MyLongNamedReactFunctionalComponent2: FunctionComponent<
  ALongNamedInterface1 | ALongNamedInterface2
> = props => {};

// functional component with valid line break
export const MyLongNamedReactFunctionalComponent3: FunctionComponent<
  ALongNamedInterface1,
  ALongNamedInterface2
> = props => {};

```
The first component in the above code should be parsed like the other two.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.11.0` |
| `TypeScript`                | `3.4.5` |
| `ESLint`                    | `6.0.1` |
| `node`                      | `10.14.1` |
| `npm`                       | `6.4.1` |

[Click here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAZgVymGAS2g0gFsAHaBGDAd0JgAsMcBzAGwE8MPCo4AIwBOcAIYBrADoCAHlWG1IUAM60AslwAy0NgDkxZOABMASuIIAxPAWJQxHAMIRK1WAEYkGa-iLRnrgKwADwAgjpQ+oYmAJKwcMJYYmBw7hgAZBjhugZGxnHwiclwAEwAfBgAvBgAFBTCEBQqAJRVFcAAvjIy6Ng2flCkLlRBtAzMGABuDoTGvPxCopIycPIQikOqGto50WYWMD620A4BIzQlXkcDZ24wYRFReQUJSSlpAD5Zj7mx8UUpcpVWr1RotNoYTrdKC9XC+OxDQI0eiMFjTPhzPgCETiaRyBRKaBqDCaH57czJQ79OynYZ3ADMV2p-jpoweu2e-zeqQANN8OX9CtygdU6g0mq1Ku0OiAeSBGgMVMhQGJhA06AAFVUIJUoMSTCCzWUgETJCRwGAAZQoyX4bGQMGEODgcqYMDIHAA6kxGHAVDaUpadYxCJNGFxkOAVEq5fwVAkYBrhGI2GQxMgkhx43KAFYqWQAIWTYHNVuiWgWGYc2ZAedklrtHDgAEUcBB4FWsy6QDbhPHhJGYFwKH6wMJCBQYMb6vwYJ7ZsxkAAOAAMctB8c9yYokfqfoSkzgxoAjm34EmwZGxCoALQCEwmY2iU+EURJlNpzs1+NkQgOp3diojYtmeR5IJmNYwGIgjzsYi5ICUcqOmIhBYmwASfigUDUMaODxgAKtBuoQd2kzOnExg0JaY4TjAoRQMYlpDk2X5wB0HRAA) for Prettier playground link.

**Comment by bradzacher** [comment]
Please file a bug with the prettier repo, as we do not control the prettier npm package.
